src/Native/System.Native/pal_errno.cpp

```diff
@@ -299,7 +303,11 @@ extern "C" int32_t SystemNative_ConvertErrorPalToPlatform(Error error)
         case PAL_ENOTEMPTY:
             return ENOTEMPTY;
         case PAL_ENOTRECOVERABLE:
+#if HAVE_ENOTRECOVERABLE
             return ENOTRECOVERABLE;
+#else
+            return -1;
+#endif
```

Comment:
I don't understand. If those values don't exist on the platform but they end up in the function, then they should trigger an assert. And if they don't exist, then they should similarly be ifdef'd out of the native enum, and there won't be a warning.